src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -154,217 +162,180 @@ def process(
         )
         bundle_map.add(artifact_to_process)
 
-        # 2. Get raw extension functions through Snowpark callback
-        dest_file_py_file_to_collected_raw_ex_fns: Dict[Path, Optional[Any]] = {}
-
-        def is_python_file_artifact(src: Path, dest: Path):
-            return src.is_file() and src.suffix == ".py"
-
-        for src_file, dest_file in bundle_map.all_mappings(
-            absolute=True, expand_directories=True, predicate=is_python_file_artifact
-        ):
-            try:
-                collected_raw_ex_fns = _execute_in_sandbox(
-                    py_file=str(dest_file.resolve()),
-                    deploy_root=self.deploy_root,
-                    kwargs=kwargs,
-                )
-            except Exception as exc:
-                cc.warning(
-                    f"Error processing extension functions in {src_file}: {exc}"
-                )  # Display the actual file for the user to inspect
-                cc.warning("Skipping generating code of all objects from this file.")
-                collected_raw_ex_fns = None
-
-            if not collected_raw_ex_fns:
-                continue
+        collected_extension_functions_by_path = self.collect_extension_functions(
+            bundle_map, processor_mapping
+        )
 
-            cc.message(f"This is the file path in deploy root: {dest_file}\n")
-            cc.message("This is the list of collected extension functions:")
-            cc.message(pprint.pformat(collected_raw_ex_fns))
+        collected_output = []
+        for py_file, extension_fns in collected_extension_functions_by_path.items():
+            for extension_fn in extension_fns:
+                create_stmt = generate_create_sql_ddl_statement(extension_fn)
+                if create_stmt is None:
+                    continue
 
-            filtered_collection = list(
-                filter(
-                    lambda item: (item is not None) and (len(item) > 0),
-                    collected_raw_ex_fns,
-                )
-            )
-            if len(filtered_collection) != len(collected_raw_ex_fns):
-                cc.warning(
-                    "Discovered extension functions that have value None or do not contain any information."
-                )
-                cc.warning(
-                    "Skipping generating code of all such objects from this file."
+                cc.message(
+                    "-- Generating Snowpark annotation SQL code for {}".format(py_file)
                 )
-
-            # 4. Enrich the raw extension functions by setting additional properties
-            for raw_ex_fn in filtered_collection:
-                sanitize_extension_function_data(ex_fn=raw_ex_fn, py_file=dest_file)
-                enrich_ex_fn(
-                    ex_fn=raw_ex_fn,
-                    py_file=dest_file,
-                    deploy_root=self.deploy_root,
+                cc.message(create_stmt)
```

Comment:
Is this intended to be in the final version, or just for debugging for now?

Reply:
Definitely temporary